### PR TITLE
[mark-scan] Handle status transmission command

### PR DESCRIPTION
**Pull request by kshen0**
## Overview
Buffering print data (or any data) to the generic transfer-out buffer may invoke the `real-time status transmission` command if the buffered data contains a byte sequence matching that command (`0x10 0x04`). This causes unsolicited data to be passed to the generic transfer-in buffer, which breaks parsing the next time the driver issues a command that responds over the generic transfer-in buffer.

See comment here: https://github.com/votingworks/vxsuite/pull/4846/files#diff-6d4416eebc285362e1f02d2f6252d35972e5c0db67a2db365540cb87a2bea431R388 and [Slack](https://votingworks.slack.com/archives/C03Q42V9MJ5/p1715296095534459) for more.

## Demo Video or Screenshot

Successful print of ballot with problematic byte sequence

https://github.com/votingworks/vxsuite/assets/1060688/7a52b519-e07a-4bba-b86c-ecf444566709

## Testing Plan

- added tests
- manually verified ballot with problematic byte sequence can be printed

## Checklist

- [ ] (N/A) I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

### CODE REVIEW

File: apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts
Comment (kshen0) [inline apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts:854]: This event is noisy and has little value in the logs

File: libs/custom-paper-handler/src/driver/driver.test.ts
Comment (kshen0) [inline libs/custom-paper-handler/src/driver/driver.test.ts:690]: As far as I can tell, the paper handler will buffer responses in discrete chunks. So when 2 commands are executed without reading the transfer in buffer after the first command, we can still expect reading the transfer in buffer to return a single response rather than mashing them all together.

File: libs/custom-paper-handler/src/driver/driver.ts
Comment (kshen0) [inline libs/custom-paper-handler/src/driver/driver.ts:552]: Found these useful so I left them in

File: libs/custom-paper-handler/src/driver/driver.ts
Comment (arsalansufi) [inline libs/custom-paper-handler/src/driver/driver.ts:423]: Is it possible that a blob might contain a _valid_ status transmission request? Would we need to handle those responses as well here?

Comment (arsalansufi) [reply libs/custom-paper-handler/src/driver/driver.ts:423]: This is already handled - @kshen0 walked me through it in person

That said, we're now taking a step back and leaning toward the likely safer alternative of just always reading/clearing the in buffer after the print command, instead of catching unexpected responses from a previous print command while reading the response for a subsequent command

File: libs/custom-paper-handler/src/driver/coders.ts
Comment (adghayes) [inline libs/custom-paper-handler/src/driver/coders.ts:140]: In what way is this different from the `PrinterStatusRealTimeExchangeResponse` above?

File: libs/custom-paper-handler/src/driver/driver.ts
Comment (adghayes) [reply libs/custom-paper-handler/src/driver/driver.ts:423]: This makes to me sense provided we aren't worried about other endpoints accidentally triggering the same command.

File: libs/custom-paper-handler/src/driver/coders.ts
Comment (kshen0) [reply libs/custom-paper-handler/src/driver/coders.ts:140]: It's the same but lacks the real-time response header. I can factor it out and use spread syntax.

Comment (adghayes) [reply libs/custom-paper-handler/src/driver/coders.ts:140]: Your call, just wanted to make sure the code duplication was intentional.

File: libs/custom-paper-handler/src/driver/driver.test.ts
Comment (kshen0) [reply libs/custom-paper-handler/src/driver/driver.test.ts:690]: Confirmed this by running `enablePrint`, `printChunk` with the bad chunk, and `setScanDirection` with 5s sleep between each. Reading the transfer in buffer after all commands are run results in 3 discrete, parsable responses of `0x06`, the `RealTimeStatusTransmission` response, and `0x06`.

File: libs/custom-paper-handler/src/driver/driver.ts
Comment (kshen0) [reply libs/custom-paper-handler/src/driver/driver.ts:423]: For the record it is possible to send a well-formed status transmission request `0x10 0x04 0x14` and we do handle it appropriately.

We looked into reading the buffer after every print command, but with the WebUSB API once a `transferInGeneric` is initiated it can't be canceled. So in the vast majority of cases we would block on the `transferInGeneric` command.

We can avoid blocking the caller with standard use of `Promise.race` as we do with `clearGenericInBuffer`. But that still initiates a `transferInGeneric` request, so the next response to a valid command over that buffer will be read and discarded by the `transferInGeneric` request that "timed out"`*`

For simplicity I believe it's best to `transferInGeneric` only when we are guaranteed a response on the transfer-in buffer, and handle unexpected extra data as it comes up.

`*` Example in more detail:

* `printChunk` is called with unproblematic data
* Initiate a `transferInGeneric` (call this `transfer in A`) that unblocks the caller after a timeout but does not (cannot) cancel the transfer in request
* After timeout, state machine continues to issue next generic command `setScanDirection('backward')`
  * This and every other generic command involves 2 steps: issuing the command over the generic transfer out buffer and reading the response over the generic transfer in buffer (call this `transfer in B`)
* After execution of `setScanDirection`, device responds on transfer in buffer with `0x06`
* `transfer in A` request resolves by reading `0x06` from the transfer in buffer. The buffer is now empty
* `transfer in B` is now blocked until the subsequent command responds over the transfer in buffer

Comment (kshen0) [reply libs/custom-paper-handler/src/driver/driver.ts:423]: > This makes to me sense provided we aren't worried about other endpoints accidentally triggering the same command.

The only command that sends arbitrary data is `bufferChunk` and its caller `printChunk`.

File: libs/custom-paper-handler/src/driver/coders.ts
Comment (kshen0) [reply libs/custom-paper-handler/src/driver/coders.ts]: Good call, [5802d37](https://github.com/votingworks/vxsuite/pull/4846/commits/5802d37fbb73a3e26c00135302c8f4a12fc44a6e)

**Review by arsalansufi — APPROVED** [review]
LGTM!

**Pull request merged**
